tsdb/errors/errors.go

```diff
@@ -79,7 +79,40 @@ func (es nonNilMultiError) Error() string {
 	return buf.String()
 }
 
-// CloseAll closes all given closers while recording error in MultiError.
+// As finds the first error in multiError slice of error chains matches target, and if so, sets
+// target to that error value and returns true. Otherwise, it returns false.
+//
+// An error matches target if the error's concrete value is assignable to the value
+// pointed to by target, or if the error has a method As(interface{}) bool such that
+// As(target) returns true. In the latter case, the As method is responsible for
+// setting target.
+func (es multiError) As(target interface{}) bool {
```

Comment:
I assumed initially that making type priv means we user can't As(multiErr), but that's not true (to verify)

< Offline 1:1 with @codesome >